### PR TITLE
objects: fix gdrive CI issue

### PR DESCRIPTION
**Pull request by pmrowla**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/fs/base.py [hunk 1]
```diff
@@ -182,7 +182,10 @@ def info(self, path_info):
         raise NotImplementedError
 
     def getsize(self, path_info):
-        return self.info(path_info).get("size")
+        size = self.info(path_info).get("size")
+        if size is not None:
+            size = int(size)
+        return size
```
Comment (efiop) [inline dvc/fs/base.py:188]: Was gdrive returning string size? `fs.info` should always return size as int, so prob worth fixing in `gdrive.py` itself.

Comment (pmrowla) [reply dvc/fs/base.py:188]: Yeah, there were 2 issues

- after the transfer change we were verifying the destination ODB on both push and pull, when previously we only verified on pull
- the verify on push caused us to read sizes from the gdrive fs, which made the issue w/gdrive returning string sizes show up

I'll move the conversion into our gdrive fs, still waiting for the branch build to finish to make sure there's nothing else

Comment (efiop) [reply dvc/fs/base.py:188]: @pmrowla Nice catch! :fire:

**Comment by efiop** [comment]
Thank you, @pmrowla ! :pray:

**Pull request merged**